README.md

```diff
@@ -1,141 +1,481 @@
+| `CH_ODBC_PREFER_BUNDLED_THIRD_PARTIES` |                           `ON`                           | Prefer bundled over system variants of third party libraries       |
+|     `CH_ODBC_PREFER_BUNDLED_POCO`      | inherits value of `CH_ODBC_PREFER_BUNDLED_THIRD_PARTIES` | Prefer bundled over system variants of Poco library                |
+|      `CH_ODBC_PREFER_BUNDLED_SSL`      |     inherits value of `CH_ODBC_PREFER_BUNDLED_POCO`      | Prefer bundled over system variants of TLS/SSL library             |
+|  `CH_ODBC_PREFER_BUNDLED_GOOGLETEST`   | inherits value of `CH_ODBC_PREFER_BUNDLED_THIRD_PARTIES` | Prefer bundled over system variants of Google Test library         |
+|    `CH_ODBC_PREFER_BUNDLED_NANODBC`    | inherits value of `CH_ODBC_PREFER_BUNDLED_THIRD_PARTIES` | Prefer bundled over system variants of nanodbc library             |
+|     `CH_ODBC_RUNTIME_LINK_STATIC`      |                          `OFF`                           | Link with compiler and language runtime statically                 |
+|   `CH_ODBC_THIRD_PARTY_LINK_STATIC`    |                           `ON`                           | Link with third party libraries statically                         |
+|       `CH_ODBC_DEFAULT_DSN_ANSI`       |                 `ClickHouse DSN (ANSI)`                  | Default ANSI DSN name                                              |
+|     `CH_ODBC_DEFAULT_DSN_UNICODE`      |                `ClickHouse DSN (Unicode)`                | Default Unicode DSN name                                           |
+|               `TEST_DSN`               |       inherits value of `CH_ODBC_DEFAULT_DSN_ANSI`       | ANSI DSN name to use in tests                                      |
+|              `TEST_DSN_W`              |     inherits value of `CH_ODBC_DEFAULT_DSN_UNICODE`      | Unicode DSN name to use in tests                                   |
+
+Configuration options above can be specified in the first `cmake` command (generation step) in a form of `-Dopt=val`.
+
+## Appendices
+
+### Run-time dependencies: Windows
+
+All modern Windows systems come with preinstalled MDAC driver manager.
+
+Another run-time dependecies are `C++ Redistributable for Visual Studio 2017` or same for `2019`, etc., depending on the package being installed, however the required DLL's are redistributed with the `.msi` installer, and you can choose to install them from there, if you don't already have them installed in your system.
+
+### Run-time dependencies: macOS
+
+#### iODBC <!-- omit in toc -->
+
+Homebrew: execute the following in the terminal (assuming you have [Homebrew](https://brew.sh/) installed):
+
+```sh
+brew install poco openssl libiodbc
+```
+
+#### UnixODBC <!-- omit in toc -->
+
+Homebrew: execute the following in the terminal (assuming you have [Homebrew](https://brew.sh/) installed):
+
+```sh
+brew install poco openssl unixodbc
+```
+
+### Run-time dependencies: Red Hat/CentOS
```

Comment:
What's about standard lib? I don't think it's linked statically.

Reply:
I don't see a need in any external package so far. Default `libstdc++.so.6` in CentOS works just fine. Maybe soon the code will rely on fresher libc++ implementations.